pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -63,25 +64,28 @@ buildDotnetModule rec {
     libX11
   ];
 
-  executables = [
-    nexusmods-app.meta.mainProgram
-  ];
+  executables = [ nexusmods-app.meta.mainProgram ];
```

Comment:
@SuperSandro2000 do you still want references to `meta.mainProgram` removed? Should I tackle that in this PR?

Reply:
My suggestion would be to just `let executable = ...` at the top.  `nexusmods-app` is not necessarily the final package we're evaluating here.

There's a similar problem with tests, where `nexusmods-app.tests` and `nexusmods-app-unfree.tests` evaluate to the same thing.

I think ideally we'd avoid referencing ourself in the `callPackage` args, but we probably need `buildDotnetModule` to support `finalAttrs` first.

Reply:
> My suggestion would be to just `let executable = ...` at the top. `nexusmods-app` is not necessarily the final package we're evaluating here.

Since we're already using a `rec` attrset, we could directly reference `meta.mainProgram` too, but that won't solve the test cases.

> I think ideally we'd avoid referencing ourself in the `callPackage` args, but we probably need `buildDotnetModule` to support `finalAttrs` first.

Agreed, I think that is needed to build the tests properly.

I'll take a quick look at adding support to `buildDornetModule`.

Reply:
The `rec` thing sounds good to me too, and we don't have to fix the tests in this PR if it turns into a big project.